config/imsim-config.yaml

```diff
@@ -0,0 +1,293 @@
+# This config file is intended to approximately reproduce what imSim does for e.g. DC2.
+# It won't match in detail, since our random number treatment is different.
+# And not all features are implemented yet, but the main functionality should be pretty close.
+
+
+# This tells GalSim which things to import to register extra config types
+# Mostly we need the imsim repo.  But could add others here as well.
+# E.g. galsim_extra has some useful routines we might want to use.  (But I don't here.)
+modules:
+    - imsim
+    - astropy  # I don't really understand why this is required.  But if only have astropy.time
+               # then I get a NameError when trying to parse astropy.time.Time.
+    - astropy.time  # Need this for an Eval.  Tell GalSim to have it loaded.
+
+# Anything set here can be used in any Eval (usually a string that starts with $) as a variable.
+# This is often convenient, so they can be set just once (probably from the command line) and
+# used in multiple places.
+eval_variables:
+    # The first letters of these variables indicates the type.
+    # s=string is the only one we have here.
+    # So to use this in an Eval string, you would write just run, not srun.
+    srun: '0001'
+
+# Any input data is set here.  These are read in at the start of the program and potentially
+# updated for each output file.
+# Also includes things that need some set up at the start of an exposure, like the atmospheric PSF.
+input:
+    camera_geometry: {}
+
+    instance_catalog:
+        # This enables InstCat types
+        file_name: {} # No default instance catalog.
```

Comment:
This strikes me as slightly confusing, since the intended value here is a string, not a dict.  If you plan to always give this on the command line, I'd suggest putting something that looks like a file_name (e.g. `instance_catalog.txt`) and a comment saying that this is normally overridden on the command line using `galsim imsim-config.yaml input.instance_catalog.file_name=/path/to/some/instance_catalog.txt`.

Reply:
OK. Good idea I changed this to make the error message more comprehensible.